StreetDrop/StreetDrop/Data/DataMapping/SearchedMusicList.swift

```diff
@@ -0,0 +1,46 @@
+//
+//  SearchedMusicList.swift
+//  StreetDrop
+//
+//  Created by 맹선아 on 2023/05/14.
+//
+
+import Foundation
+
+struct SearchedMusicList: Codable {
+    let list: [Music]
+
+    struct Music: Codable {
+        let albumName: String
+        let artistName: String
+        let songName: String
+        let durationTime: String
+        let albumImage: URL
+        let albumThumbnailImage: URL
+
+        private enum CodingKeys: String, CodingKey {
+            case albumName, artistName, songName, durationTime
+            case albumImage = "albumImg"
+            case albumThumbnailImage = "albumThumbnailImg"
+        }
+
+        init(from decoder: Decoder) throws {
+            let container = try decoder.container(keyedBy: CodingKeys.self)
+            self.albumName = try container.decode(String.self, forKey: .albumName)
+            self.artistName = try container.decode(String.self, forKey: .artistName)
+            self.songName = try container.decode(String.self, forKey: .songName)
+            self.durationTime = try container.decode(String.self, forKey: .durationTime)
+            self.albumImage = try container.decode(URL.self, forKey: .albumImage)
+            self.albumThumbnailImage = try container.decode(URL.self, forKey: .albumThumbnailImage)
+        }
+    }
+
+    private enum CodingKeys: String, CodingKey {
+        case list = "data"
+    }
+
+    init(from decoder: Decoder) throws {
+        let container = try decoder.container(keyedBy: CodingKeys.self)
+        self.list = try container.decode([SearchedMusicList.Music].self, forKey: .list)
+    }
+}
```

Comment:
코드양을 줄이기 위해, `init(from decoder: Decoder)`을 필요시에 사용하면 좋을 것 같습니다!

예시 코드입니다!
```suggestion
   struct SearchMusicResponseDTO: Decodable {
    let musicList: [Music]

    struct Music: Decodable {
        let albumName: String
        let artistName: String
        let songName: String
        let durationTime: String
        let albumImage: URL
        let albumThumbnailImage: URL

        private enum CodingKeys: String, CodingKey {
            case albumName, artistName, songName, durationTime
            case albumImage = "albumImg"
            case albumThumbnailImage = "albumThumbnailImg"
        }
    }

    private enum CodingKeys: String, CodingKey {
        case musicList = "data"
    }
}
```

Reply:
너무너무 좋은 것 같습니다! 
수정해서 커밋했습니다 :) !! ([5d0642f](https://github.com/depromeet/street-drop-iOS/pull/22/commits/5d0642f46c56298ff2991bf9d718d79cb77ca550))
맵마커용 API에서도 똑같이 제거하도록 하겠습니다

Reply:
```swift                
switch result {
    case .success(let data):
        guard let dto = try? JSONDecoder().decode(SearchMusicResponseDTO.self, from: data) else {
            throw RepositoryError.decodingError(toType: SearchMusicResponseDTO.self)
        }
        return dto.toDomain()
    case .failure(let error):
        throw error
}
```
이런 식으로 디코딩 시킬 때의 에러는 레파지토리에서 잡아도 좋을거같아요!